core/test/nuxeo-audit-page-provider.test.js

```diff
@@ -0,0 +1,212 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+import { html, fixture, fakeServer } from '@nuxeo/testing-helpers';
+import '../nuxeo-audit-page-provider.js';
+import moment from '@nuxeo/moment';
+
+const documentId = 'myDocumentId';
+const events = ['documentCreated', 'documentModified'];
+const eventDocumentCategory = 'eventDocumentCategory';
+const user = 'Administrator';
+const from = moment().format('YYYY-MM-DD');
+const to = moment()
+  .add(5, 'days')
+  .format('YYYY-MM-DD');
+const requestParameters = {
+  startEventDate: from,
+  endEventDate: to,
+  category: eventDocumentCategory,
+  eventId: events,
+  principalName: user,
+};
+const providerParameters = {
+  startDate: from,
+  endDate: to,
+  eventCategory: eventDocumentCategory,
+  eventIds: events,
+  principalName: user,
+};
+const providerSort = {
+  eventDate: 'asc',
+  category: 'desc',
+  eventId: 'desc',
+  principalName: 'asc',
+};
+let server;
+async function getProvider(params = {}, docId = documentId, sort = {}) {
+  const provider = await fixture(
+    html`
+      <nuxeo-audit-page-provider
+        id="provider"
+        doc-Id="${docId}"
+        params="${JSON.stringify(params)}"
+        sort="${JSON.stringify(sort)}"
+        page-size="40"
+      ></nuxeo-audit-page-provider>
+    `,
+  );
+  return provider;
+}
+
+function mockDocHistoryResponse(qParams = {}) {
+  // build the expected parameters (add the commons properties)
+  const expectedQueryParam = { currentPageIndex: 0, pageSize: 40, ...qParams };
+  server.respondWith('GET', `/api/v1/id/${documentId}/@audit`, ({ queryParams }) => {
+    expect(expectedQueryParam).to.eql(queryParams);
```

Comment:
This is our mock response to the request. I wouldn't perform expects/asserts here.
If we don't need any specific logic to generate our response, I would go simply with: 
```suggestion
    server.respondWith('GET', `/api/v1/id/${documentId}/@audit`, { 'entity-type': 'logEntries', entries: [] })
```